libclamav/scanners.c

```diff
@@ -5261,6 +5261,10 @@ static cl_error_t scan_common(cl_fmap_t *map, const char *filepath, const char *
     ctx.virname = virname;
     ctx.scanned = scanned;
     ctx.options = malloc(sizeof(struct cl_scan_options));
+    if (NULL == ctx.options){
+        status = CL_EMEM;
+        goto done;
+    }
```

Comment:
Nice! Though would you want to use the `MALLOC()` macro?

PS: I forgot about those macros when working on the allmatch stuff.  Feel free to heckle me about it on that PR and I can go clean it up.